pkg/routeagent_driver/handlers/ovn/gateway_route_controller.go

```diff
@@ -0,0 +1,140 @@
+	gatewayRouteWatcher watcher.Interface
+	connectionHandler   *ConnectionHandler
+	remoteSubnets       sets.Set[string]
+	stopCh              chan struct{}
+	mgmtIP              string
+}
+
+func NewGatewayRouteController(config watcher.Config, connectionHandler *ConnectionHandler,
+	namespace string,
+) (*GatewayRouteController, error) {
+	var err error
+
+	controller := &GatewayRouteController{
+		connectionHandler: connectionHandler,
+		remoteSubnets:     sets.New[string](),
+	}
+
+	config.ResourceConfigs = []watcher.ResourceConfig{
+		{
+			Name:         "GatewayRoute watcher",
+			ResourceType: &submarinerv1.GatewayRoute{},
+			Handler: watcher.EventHandlerFuncs{
+				OnCreateFunc: controller.gatewayRouteCreatedOrUpdated,
+				OnUpdateFunc: controller.gatewayRouteCreatedOrUpdated,
+				OnDeleteFunc: controller.gatewayRouteDeleted,
+			},
+			SourceNamespace: namespace,
+		},
+	}
+
+	controller.gatewayRouteWatcher, err = watcher.New(&config)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating resource watcher")
+	}
+
+	mgmtIP, err := getNextHopOnK8sMgmtIntf()
+	if err != nil {
+		return nil, err
+	}
+
+	controller.mgmtIP = mgmtIP
+
+	err = controller.gatewayRouteWatcher.Start(controller.stopCh)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error starting the resource wather")
+	}
+
+	logger.Infof("Started GatewayRouteController")
+
+	return controller, nil
+}
+
+func (g *GatewayRouteController) gatewayRouteCreatedOrUpdated(obj runtime.Object, _ int) bool {
+	subMGWRoute := obj.(*submarinerv1.GatewayRoute)
+
+	err := g.reconcileRemoteSubnets(subMGWRoute, true)
+	if err != nil {
+		logger.Errorf(err, "Error creating or updating router policies and static routes for remote subnet %q", g.remoteSubnets)
+		return true
+	}
+
+	return false
+}
+
+func (g *GatewayRouteController) gatewayRouteDeleted(obj runtime.Object, _ int) bool {
+	subMGWRoute := obj.(*submarinerv1.GatewayRoute)
+
+	err := g.reconcileRemoteSubnets(subMGWRoute, false)
+	if err != nil {
+		logger.Errorf(err, "Error deleting router policies and static routes for remote subnet %q", g.remoteSubnets)
+		return true
+	}
+
+	return false
+}
+
+func (g *GatewayRouteController) reconcileRemoteSubnets(subMGWRoute *submarinerv1.GatewayRoute, addSubnet bool) error {
+	if len(subMGWRoute.RoutePolicySpec.NextHops) == 0 {
+		logger.Warningf("The GatewayRoute does not have next hop %v", subMGWRoute)
+		return nil
+	}
+
+	if subMGWRoute.RoutePolicySpec.NextHops[0] != g.mgmtIP {
+		return nil
+	}
+
+	for _, subnet := range subMGWRoute.RoutePolicySpec.RemoteCIDRs {
+		if addSubnet {
+			g.remoteSubnets.Insert(subnet)
+		} else {
+			g.remoteSubnets.Delete(subnet)
+		}
+	}
+
+	err := g.connectionHandler.reconcileSubOvnLogicalRouterPolicies(g.remoteSubnets, g.mgmtIP)
+	if err != nil {
+		return err
+	}
+
+	err = g.connectionHandler.reconcileOvnLogicalRouterStaticRoutes(g.remoteSubnets, g.mgmtIP)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
Nit - this could be streamlined.

```suggestion
	return g.connectionHandler.reconcileOvnLogicalRouterStaticRoutes(g.remoteSubnets, g.mgmtIP)
```

Reply:
Done.